modules/adxcgBidAdapter.js

```diff
@@ -1,410 +1,363 @@
-            }
+          if (isStr(bidResponse.ext.agency_name)) {
+            deepSetValue(result, 'meta.agencyName', bidResponse.ext.agency_name);
           }
-          bidsAll.push(bid)
-        })
-      })
-      return bidsAll
-    },
+        }
+        if (mediaType === BANNER) {
+          result.ad = bidResponse.adm;
+        } else if (mediaType === NATIVE) {
+          result.native = parseNative(bidResponse);
+          result.width = 0;
+          result.height = 0;
+        } else if (mediaType === VIDEO) {
+          result.vastUrl = bidResponse.nurl;
+          result.vastXml = bidResponse.adm;
+        }
 
-  onBidWon: (bid) => {
-    if (bid.burl) {
-      triggerPixel(replaceAuctionPrice(bid.burl, bid.originalCpm));
-    }
+        return result;
+      }
+    }).filter(Boolean);
   },
+  getUserSyncs: (syncOptions, responses, gdprConsent, uspConsent) => {
+    const syncs = [];
+    let syncUrl = config.getConfig('adxcg.usersyncUrl');
+
+    let query = [];
+    if (syncOptions.pixelEnabled && syncUrl) {
+      if (gdprConsent) {
+        query.push('gdpr=' + (gdprConsent.gdprApplies & 1));
+        query.push('gdpr_consent=' + encodeURIComponent(gdprConsent.consentString || ''));
+      }
+      if (uspConsent) {
+        query.push('us_privacy=' + encodeURIComponent(uspConsent));
+      }
 
-  onTimeout(timeoutData) {
-    if (timeoutData == null) {
-      return;
+      syncs.push({
+        type: 'image',
+        url: syncUrl + (query.length ? '?' + query.join('&') : '')
+      });
     }
-
-    let beaconParams = {
-      A: timeoutData.bidder,
-      bid: timeoutData.bidId,
-      a: timeoutData.adUnitCode,
-      cn: timeoutData.timeout,
-      aud: timeoutData.auctionId,
-    };
-    let adxcgRequestUrl = buildUrl({
-      protocol: 'https',
-      hostname: 'hbps.adxcg.net',
-      pathname: '/event/timeout.gif',
-      search: beaconParams
-    });
-    logWarn(BIDDER_CODE + ': onTimeout called');
-    triggerPixel(adxcgRequestUrl);
+    return syncs;
   },
-
-  getUserSyncs: function (syncOptions, serverResponses, gdprConsent) {
-    let params = '';
-    if (gdprConsent && 'gdprApplies' in gdprConsent) {
-      if (gdprConsent.consentString) {
-        if (typeof gdprConsent.gdprApplies === 'boolean') {
-          params += `?gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
-        } else {
-          params += `?gdpr=0&gdpr_consent=${gdprConsent.consentString}`;
-        }
-      }
-    }
-
-    if (syncOptions.iframeEnabled) {
-      return [{
-        type: 'iframe',
-        url: 'https://cdn.adxcg.net/pb-sync.html' + params
-      }];
+  onBidWon: (bid) => {
+    // for native requests we put the nurl as an imp tracker, otherwise if the auction takes place on prebid server
+    // the server JS adapter puts the nurl in the adm as a tracking pixel and removes the attribute
+    if (bid.nurl) {
+      ajax(replaceAuctionPrice(bid.nurl, bid.originalCpm))
```

Comment:
I'm not sure if a bid adapter is allowed to make an Ajax request apart from an Auction request. Is there any way you can avoid it?